computer_vision/richardson_lucy.py

```diff
@@ -0,0 +1,241 @@
+        >>> kernel = np.array([[1, 0], [0, 1]])
+        >>> convolve(matrix, kernel)
+        array([[ 0.07841354, -0.14142533],
+               [-0.14142533,  0.07841354]])
+    """
+    if kernel.shape[0] > matrix.shape[1] or kernel.shape[1] > matrix.shape[1]:
+        return matrix
+    kernel = pad_to_size(kernel, matrix)
+
+    kernel_f = fft2(kernel)
+    matrix_f = fft2(matrix)
+
+    return ifftshift(ifft2(np.multiply(matrix_f, kernel_f))).real
+
+
+def get_motion_psf(shape: tuple, angle: float, num_pixel_dist: int = 20) -> np.ndarray:
+    """
+    Generate an array with given shape corresponding to
+    Point Spread Function for the desired angle.
+
+    Args:
+        shape: The shape of the image.
+        angle: The angle of the motion blur. Should be in degrees. [0, 360)
+        num_pixel_dist: The distance of the motion blur. [0, infinity)
+                        Remember that the distance is measured in pixels.
+                        Greater values will be more blurry
+
+    Returns:
+        np.ndarray: The point-spread array associated with the motion blur.
+
+    Examples:
+        >>> shape = (3, 3)
+        >>> angle = 15
+        >>> get_motion_psf(shape, angle, num_pixel_dist=3)
+        array([[0.        , 0.33333333, 0.        ],
+               [0.        , 0.33333333, 0.        ],
+               [0.33333333, 0.        , 0.        ]])
+    """
+    psf = np.zeros(shape)
+    center = np.array([shape[0] - 1, shape[1] - 1]) // 2
+    radians = angle / 180 * np.pi
+    phase = np.array([np.cos(radians), np.sin(radians)])
+    for i in range(num_pixel_dist):
+        offset_x = int(center[0] - np.round_(i * phase[0]))
+        offset_y = int(center[1] - np.round_(i * phase[1]))
+        psf[offset_x, offset_y] = 1
+    psf /= psf.sum()
+
+    return psf
+
+
+def richardson_lucy(
+    degraded: np.ndarray, function_kernel: np.ndarray, steps: int
+) -> np.ndarray:
+    """
+    Richardson-Lucy method to restore an image
+    affected by known motion blur function, as well as arbitrary steps
+    to perform during iterative image restoration.
+
+    Args:
+        degraded: Observed image, considered to be degraded - to be restored
+        function_kernel: Supposed function used to blur original image
+        steps: Iterative steps to take for restoring image
+
+    Returns:
+        np.ndarray: Restored image after method application
+
+
+    Examples:
+        >>> np.random.seed(0)
+        >>> shape = (3, 3)
+        >>> degraded = gaussian_noise(shape)
+        >>> kernel = np.identity(2)
+        >>> richardson_lucy(degraded, kernel, steps=4)
+        array([[1.16272775e+02, 0.00000000e+00, 7.58091075e+01],
+               [2.40582123e+01, 2.09006034e+01, 0.00000000e+00],
+               [0.00000000e+00, 1.52620152e-02, 1.31734948e+00]])
+    """
+    estimated_img = np.full(shape=degraded.shape, fill_value=1, dtype="float64")
+
+    for i in range(steps):
```

Comment:
```suggestion
    for _ in range(steps):
```
`i` is never used in the loop body, and this is causing the pre-commit to fail